### PR TITLE
Make tabs fit to their container, even when scrollable.

### LINKED ISSUE #163: Scrollable tabs in large containers
### Description
Using scrollable paper-tabs does not play well with responsive usage since they fail to property expand to take the available space prior to there being enough that they need to scroll.

### Expected outcome
Tabs should expand to fit the container

### Actual outcome
Tabs do not expand to fit the container

### Live Demo
https://jsbin.com/kofamuzemo/edit?html,output

### Steps to reproduce
1. Add a `<paper-tabs scrollable>` element with some `<paper-tab>`s to a page
2. Have  a large enough container that tabs do not need to scroll 

### Browsers Affected
All

### PR DESCRIPTION
**Pull request by sparhami**
This fixes #163 by making the tabsContent use flex in the scrollable mode.

### CODE REVIEW

**Comment by sparhami** [comment]
Tested on IE10, IE11, Firefox and Chrome

Before: https://jsbin.com/jexegexomi/1/edit?html,output
After: https://jsbin.com/japaqopiyu/1/edit?html,output

**Comment by bicknellr** [comment]
Thanks for the PR; I think we should put this feature behind a flag because the current behavior is intentional ([see 'Desktop tabs' here](https://www.google.com/design/spec/components/tabs.html)) and changing this default could break many users' layouts.

File: paper-tabs.html [hunk 1]
```diff
@@ -133,15 +133,19 @@
...
+        min-width: 100%;
         white-space: nowrap;
       }
 
-      #tabsContent.horizontal {
-        @apply(--layout-horizontal);
+      #tabsContent.scrollable > ::content > * {
+        -ms-flex-shrink: 0;
+        -webkit-flex-shrink: 0;
+        flex-shrink: 0;
```
Comment (bicknellr) [inline paper-tabs.html]: Any particular reason we need `flex-shrink: 0`? Does the default `flex-basis: auto` not prevent the items from collapsing too much here?

Comment (sparhami) [inline paper-tabs.html]: It works fine without it in Chrome, but it is needed for IE10 (IIRC IE11 as well, can't remember if it was needed for Edge). Could just make it `-ms-flex-shrink`.

**Comment by sparhami** [comment]
Updated to put it behind a property for now. Perhaps instead the expanding behavior should be controlled by a media query?

**Comment by bicknellr** [comment]
Yeah, using a media query would probably have cleanly handled more use cases but we can't break our current users to fix it. (Per-spec accessibility is implemented as the `autoselect` flag for the same reason. :/ )

I had to play around a bit with this PR since it seems to cause scrollable tabs which can't fit in their container to be squished in IE10/11. From what I can tell, IE10 doesn't support the long form `flex-*` properties and 11 dropped support for the `-ms-` prefixed version:
```diff
diff --git a/paper-tabs.html b/paper-tabs.html
index f2587e3..f779a4a 100644
--- a/paper-tabs.html
+++ b/paper-tabs.html
@@ -133,7 +133,6 @@ Custom property | Description | Default
         -moz-flex-basis: auto;
         -ms-flex-basis: auto;
         flex-basis: auto;
-        @apply(--layout-horizontal);
       }
 
       #tabsContent.scrollable {
@@ -141,13 +140,20 @@ Custom property | Description | Default
         white-space: nowrap;
       }
 
-      .fit-container {
+      #tabsContent:not(.scrollable),
+      #tabsContent.scrollable.fit-container {
+        @apply(--layout-horizontal);
+      }
+
+      #tabsContent.scrollable.fit-container {
         min-width: 100%;
       }
 
-      #tabsContent.scrollable > ::content > * {
+      #tabsContent.scrollable.fit-container > ::content > * {
         /* IE - prevent tabs from compressing when they should scroll. */
-        -ms-flex-shrink: 0;
+        -ms-flex: 1 0 auto;
+        -webkit-flex: 1 0 auto;
+        flex: 1 0 auto;
       }
 
       .hidden {
@@ -401,7 +407,7 @@ Custom property | Description | Default
       },
 
       _computeTabsContentClass: function(scrollable, fitContainer) {
-        return scrollable ? 'scrollable' : '' + fitContainer ? ' fit-container' : '';
+        return scrollable ? 'scrollable' + (fitContainer ? ' fit-container' : '') : ' fit-container';
       },
 
       _computeSelectionBarClass: function(noBar, alignBottom) {
```

**Comment by sparhami** [comment]
Added your changes: Here is an updated bin showing the current status:  https://jsbin.com/higequyoco/edit?html,output

**Comment by bicknellr** [comment]
Thanks, looks good, trying the tests out again to see if we get better luck with Firefox + Selenium. I tested it manually though and it seems fine.

**Comment by bicknellr** [comment]
I've manually tested this with WCT over all the full set of browsers and everything still checks out, so I'm going to merge.

**Comment by sparhami** [comment]
Awesome, thanks!

**Pull request merged**